lib/decorators/placeholders.js

```diff
@@ -157,31 +157,39 @@ function isComponentPropEmpty(field) {
  */
 export function hasPlaceholder(uri, path, storeOverride = store) {
   const group = get(uri, path),
-    fields = group.fields,
-    schema = group.schema,
+    schema = group ? group.schema : null,
     placeholder = schema && schema[placeholderProp],
-    isPermanentPlaceholder = placeholder && !!placeholder.permanent,
-    isField = schema && !!schema[fieldProp],
-    isGroup = schema && !!schema.fields,
-    isComponentList = schema && !!schema[componentListProp],
-    isCollapsibleList = isComponentList && !!_.get(schema, `${componentListProp}.collapse`),
-    isComponentProp = schema && !!schema[componentProp];
+    isPermanentPlaceholder = placeholder && !!placeholder.permanent;
 
   // if there isn't a placeholder at all, return negative quickly
   // since we're ususally checking a bunch of fields/groups
-  if (!placeholder) {
-    return false;
-  }
-
   // if it has a placeholder...
   // if it's a permanent placeholder, it always displays
   // if it's a field, make sure it's empty
   // if it's a group, make sure it points to an empty field
   // if it's a component list, make sure it's empty
   // if it's a component prop, make sure it's empty
-  if (isPermanentPlaceholder) {
+  if (!placeholder) {
+    return false;
+  } else if (isPermanentPlaceholder) {
     return true;
-  } else if (isField) {
+  } else {
+    return returnPlaceHolder(uri, path, storeOverride);
+  }
+}
+
+function returnPlaceHolder(uri, path, storeOverride = store) {
+  const group = get(uri, path),
+    fields = group ? group.fields : null,
+    schema = group ? group.schema : null,
```

Comment:
We can avoid this ternary condition if we set a fallback value for `group`
```js
const group = get(uri, path) || {},
    { fields = null, schema = null } = group,
```

Reply:
👍